src/Framework/Framework/Compilation/ViewCompiler/DefaultViewCompilerCodeEmitter.cs

```diff
@@ -0,0 +1,360 @@
+        {
+            var argumentTypes = arguments.Select(a => a.Type).ToArray();
+            var constructor = type.GetConstructor(argumentTypes);
+
+            return Expression.New(constructor, arguments.ToArray());
+        }
+
+        public static Expression EmitCreateArray(Type elementType, IEnumerable<Expression> values)
+        {
+            //new [elementType] [] = { ([elementType])v1, ([elementType])v2, ([elementType])v3, ... }
+            //note: [elementType] is name of the type provided in 'elementType' parameter.
+
+            var convertedValues = values.Select(v => Expression.Convert(v, elementType));
+            return Expression.NewArrayInit(elementType, convertedValues);
+        }
+
+        /// <summary>
+        /// Emits the control builder invocation.
+        /// </summary>
+        public ParameterExpression EmitInvokeControlBuilder(Type controlType, string virtualPath)
+        {
+            var builderName = "c" + CurrentControlIndex + "_builder";
+            var untypedName = "c" + CurrentControlIndex + "_untyped";
+            var name = "c" + CurrentControlIndex;
+            CurrentControlIndex++;
+
+            //var [builderName] = controlBuilderFactory.GetControlBuilder(virtualPath).Item2.Value
+
+            var controlBuilderFactoryParameter = GetParameterOrVariable(ControlBuilderFactoryParameterName);
+
+            var getBuilderCall = Expression.Call(controlBuilderFactoryParameter, nameof(IControlBuilderFactory.GetControlBuilder), emptyTypeArguments, EmitValue(virtualPath));
+
+            var builderValueExpression = Expression.PropertyOrField(
+                Expression.PropertyOrField(getBuilderCall, "Item2"),
+                "Value");
+
+            var builderParameter = EmitCreateVariable(builderValueExpression);
+
+            //var [untypedName] = [builderName].BuildControl(controlBuilderFactory, services)
+
+            var servicesParameter = GetParameterOrVariable(ServiceProviderParameterName);
+
+            var buildControlCall = Expression.Call(builderParameter, nameof(IControlBuilder.BuildControl), emptyTypeArguments, controlBuilderFactoryParameter, servicesParameter);
+
+            return EmitCreateVariable(Expression.Convert(buildControlCall, controlType));
+        }
+
+        /// <summary>
+        /// Emits the set property statement.
+        /// </summary>
+        public void EmitSetProperty(string controlName, string propertyName, Expression valueExpression)
+        {
+            //[controlName].[propertyName] = [value] 
+            var controlParameter = GetParameterOrVariable(controlName);
+            var assigment = Expression.Assign(Expression.PropertyOrField(controlParameter, propertyName), valueExpression);
+
+            blockStack.Peek().Expressions.Add(assigment);
+        }
+
+        public Expression CreateDotvvmPropertyIdentifier(DotvvmProperty property)
```

Comment:
I think we can inline this method (and other very simple ones)

Reply:
Removed, replaced by `EmitValue`. The methot also no longer creates any identifier so it was at best unhelpful. ;)